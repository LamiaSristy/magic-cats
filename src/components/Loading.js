import React from 'react';
import Loader from 'react-loader-spinner';
import 'react-loader-spinner/dist/loader/css/react-spinner-loader.css';
import '../stylesheets/Loading.scss';
import Navbar from './Navbar';

const Loading = () => (
  <div className="loading-wrap">
    <Navbar />
    <div className="loading">
      <Loader
        type="TailSpin"
        color="#aaa"
      />
    </div>
  </div>
);

export default Loading;
