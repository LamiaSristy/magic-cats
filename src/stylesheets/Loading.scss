.loading {
  background-color: #333;
  min-height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
}
