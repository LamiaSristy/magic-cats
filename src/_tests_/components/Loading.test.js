import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom/extend-expect';
import Loading from '../../components/Loading';

const setup = () => shallow(<Loading />);

describe('Loading component', () => {
  let component;

  beforeEach(() => {
    component = setup();
  });

  it('Should render', () => {
    const element = component.find('.loading-wrap');
    expect(element.length).toEqual(1);
  });

  test('renders the loading-wrap in the page', () => {
    const item = component.find('.loading-wrap');
    expect(item).toMatchSnapshot();
  });
 
});
