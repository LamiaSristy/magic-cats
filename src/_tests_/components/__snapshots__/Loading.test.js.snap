// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component renders the cat-images the page 1`] = `ShallowWrapper {}`;

exports[`Loading component renders the loading in the page 1`] = `ShallowWrapper {}`;

exports[`Loading component renders the loading-wrap in the page 1`] = `ShallowWrapper {}`;

exports[`Loading component renders the loading-wrap in the page 2`] = `ShallowWrapper {}`;
